vscode-dotnet-runtime-library/src/Utils/WebRequestWorker.ts

```diff
@@ -58,4 +65,16 @@ export class WebRequestWorker {
     protected async cacheResults(response: string) {
         await this.extensionState.update(this.extensionStateKey, response);
     }
+
+    private async makeWebRequestWithRetries(throwOnError: boolean, retriesCount: number): Promise<any> {
+        for (let i = 0; i < retriesCount; i++) {
+            try {
+                return await this.makeWebRequest(throwOnError);

```

Comment:
nit: we only retire 2 twice this maybe fine. But ideally you want to do exponential back off. It can reduce service load. Maybe there is an library to do that.